providers/implementations/exchange/ecdh_exch.c

```diff
@@ -561,6 +563,23 @@ int ecdh_plain_derive(void *vpecdhctx, unsigned char *secret,
 
     ppubkey = EC_KEY_get0_public_key(pecdhctx->peerk);
 
+#ifdef FIPS_MODULE
+    bn_ctx = BN_CTX_new_ex(ossl_ec_key_get_libctx(privk));
```

Comment:
This is not a fast operation, so I would not like this running twice.
So in that case we may need to have a settable flag that gets passed here that indicates if the test has been run or not.

In the case where the test has already run, the validate test is inside the FIPS boundary,